include/NeoFOAM/fields/boundaryFields.hpp

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <Kokkos_Core.hpp>
+#include <iostream>
+#include "NeoFOAM/primitives/label.hpp"
+#include "NeoFOAM/primitives/scalar.hpp"
+#include "Field.hpp"
+
+#include "NeoFOAM/core/executor/executor.hpp"
+
+namespace NeoFOAM
+{
+// enum class boundaryType used in a kokkos view to categorize boundary
+// types
+
+
+template<typename T>
+class BoundaryFields
+{
+    /**
+     * @class boundaryFields
+     * @brief Represents the boundary fields for a computational domain.
+     *
+     * The boundaryFields class stores the boundary conditions and related information for a computational domain.
+     * It provides access to the computed values, reference values, value fractions, reference gradients, boundary types,
+     * offsets, and the number of boundaries and boundary faces.
+     */
+
+public:
+
+    //
+    // enum class boundaryTypes: int
+    // {
+    //     Dirichlet,
+    //     Neumann,
+    //     Robin
+    // };
+    /**
+     * @brief Copy constructor.
+     * @param rhs The boundaryFields object to be copied.
+     */
+
+    BoundaryFields(const BoundaryFields<T>& rhs)
+        : exec_(rhs.exec_),
+          value_(rhs.value_),
+          refValue_(rhs.refValue_),
+          valueFraction_(rhs.valueFraction_),
+          refGrad_(rhs.refGrad_),
+          boundaryTypes_(rhs.boundaryTypes_),
+          offset_(rhs.offset_),
+          nBoundaries_(rhs.nBoundaries_),
+          nBoundaryFaces_(rhs.nBoundaryFaces_)
+    {
+    }
+
+
+    BoundaryFields(const Executor& exec, int nBoundaryFaces, int nBoundaries)
+        : exec_(exec),
+          value_(exec, nBoundaryFaces),
+          refValue_(exec, nBoundaryFaces),
+          valueFraction_(exec, nBoundaryFaces),
+          refGrad_(exec, nBoundaryFaces),
+          boundaryTypes_(exec, nBoundaries),
+          offset_(exec, nBoundaries + 1),
+          nBoundaries_(nBoundaries),
+          nBoundaryFaces_(nBoundaryFaces)
+    {
+    }
+
+    /**
+     * @brief Get the view storing the computed values from the boundary condition.
+     * @return The view storing the computed values.
+     */
+    const NeoFOAM::Field<T>& value() const
+    {
+        return value_;
+    }
+
+    NeoFOAM::Field<T>& value()
```

Comment:
Maybe we can use copydoc for the missing docstrings.
```suggestion
   /** @copydoc const NeoFOAM::Field<T>& value() const
   */
    NeoFOAM::Field<T>& value()
```